allauth/account/forms.py

```diff
@@ -30,6 +30,16 @@
     from django.utils.importlib import import_module
 
 
+class PasswordVerificationMixin(object):
+    def clean(self):
+        cleaned_data = super(PasswordVerificationMixin, self).clean()
+        password1 = cleaned_data.get('password1', None)
+        password2 = cleaned_data.get('password2', None)
+        if (password1 and password2) and password1 != password2:
+            raise forms.ValidationError(_("You must type the same password each time."))
```

Comment:
👍  for making the code more DRY and putting this in a mixin. These changes do change the location of where the error is displayed (from password2 field level to form level). We should use `add_error` here and add this error to the password2 field.